src/lib/QueryRendererHelper.ts

```diff
@@ -32,10 +31,9 @@ export function explainResults(source: string): string {
     const tasksBlockQuery = new Query({ source });
 
     if (!tasksBlockQuery.ignoreGlobalQuery) {
-        const globalQuery: IQuery = new Query(getGlobalQuerySource());
-
-        if (globalQuery.source.trim() !== '') {
-            result += `Explanation of the global query:\n\n${globalQuery.explainQuery()}\n`;
+        const globalQuery = GlobalQuery.getInstance();
+        if (!globalQuery.isEmpty()) {
```

Comment:
This is also a change in behaviour.

I saw that the previous code trimmed the globalQuery text, and the new code does not trim.

So I tried setting the Global Query to 10 spaces.

I then tried this query:

~~~
```tasks
explain
limit 10
```
~~~

In Tasks 4.6.0, the explanation is:

```text
Only tasks containing the global filter '#task'.

Explanation of this Tasks code block query:

No filters supplied. All tasks will match the query.

At most 10 tasks.
```

With the code in this PR, the explanation is:

```text
Only tasks containing the global filter '#task'.

Explanation of the global query:

No filters supplied. All tasks will match the query.
Explanation of this Tasks code block query:

No filters supplied. All tasks will match the query.

At most 10 tasks.
```

Reply:
Fixed with a test

Reply:
I can't see the test, but I tested this out by building the plugin and running it... This confirmed that the behaviour is fixed. Thanks.